Source/Scene/Model.js

```diff
@@ -3354,28 +3366,28 @@ define([
         var context = frameState.context;
         var scene3DOnly = frameState.scene3DOnly;
 
-        // Retain references to updated source shaders and programs for rebuilding as needed
-        var programs = model._sourcePrograms = model.gltf.programs;
-        var shaders = model._sourceShaders = model.gltf.shaders;
+        // Retain references to updated source shaders for rebuilding as needed
+        var shaders = model._sourceShaders;
         model._hasPremultipliedAlpha = hasPremultipliedAlpha(model);
 
         var quantizedVertexShaders = model._quantizedVertexShaders;
         var toClipCoordinates = model._toClipCoordinatesGLSL = {};
-        for (var id in programs) {
-            if (programs.hasOwnProperty(id)) {
-                var program = programs[id];
+        var programs = model._sourcePrograms;
+        for (var programId in programs) {
+            if (programs.hasOwnProperty(programId)) {
+                var program = programs[programId];
                 var shader = shaders[program.vertexShader].extras._pipeline.source;
                 if (model.extensionsUsed.WEB3D_quantized_attributes || model._dequantizeInShader) {
-                    var quantizedVS = quantizedVertexShaders[id];
+                    var quantizedVS = quantizedVertexShaders[programId];
                     if (!defined(quantizedVS)) {
-                        quantizedVS = modifyShaderForQuantizedAttributes(shader, id, model);
-                        quantizedVertexShaders[id] = quantizedVS;
+                        quantizedVS = modifyShaderForQuantizedAttributes(shader, programId, model);
+                        quantizedVertexShaders[programId] = quantizedVS;
                     }
                     shader = quantizedVS;
                 }
 
-                shader = modifyShader(shader, id, model._vertexShaderLoaded);
-                toClipCoordinates[id] = ModelUtility.toClipCoordinatesGLSL(model.gltf, shader);
+                shader = modifyShader(shader, programId, model._vertexShaderLoaded);
+                toClipCoordinates[programId] = ModelUtility.toClipCoordinatesGLSL(model.gltf, shader);
             }
         }
 
```

Comment:
Right below - `model.gltf.glExtensionsUsed` - this is now part of the techniques_webgl extension.